Formula/gap.rb

```diff
@@ -0,0 +1,56 @@
+class Gap < Formula
+  desc "System for computational discrete algebra"
+  homepage "https://www.gap-system.org/"
+  url "https://www.gap-system.org/pub/gap/gap-4.10/tar.bz2/gap-4.10.0.tar.bz2"
+  sha256 "2dc71364b7418d16f8b99ca914971996fa4416294fa3ae8fb336c873c946abb6"
+
+  depends_on "gmp"
+  # GAP cannot be built against the native OSX version of readline
+  # it requires either GNU readline, or no readline at all; but
+  # the latter leads to an inferior user experience.
+  # So we depend on GNU readline here.
+  depends_on "readline"
+
+  def install
+    # XXX:  Currently there is no `install` target in `Makefile`.
+    #   According to the manual installation instructions in
+    #
+    #     https://github.com/gap-system/gap/blob/master/INSTALL.md
+    #
+    #   the compiled "bundle" is intended to be used "as is," and there is
+    #   no instructions for how to remove the source and other unnecessary
+    #   files after compilation.  Moreover, the content of the
+    #   subdirectories with special names, such as `bin` and `lib`, is not
+    #   suitable for merging with the content of the corresponding
+    #   subdirectories of `/usr/local`.  The easiest temporary solution seems
+    #   to be to drop the compiled bundle into `<prefix>/libexec` and to
+    #   create a symlink `<prefix>/bin/gap` to the startup script.
+    #   This use of `libexec` seems to contradict Linux Filesystem Hierarchy
+    #   Standard, but is recommended in Homebrew's "Formula Cookbook."
+
+    libexec.install Dir["*"]
+
+    # GAP does not support "make install" so it has to be compiled in place
+
+    cd libexec do
+      system "./configure", "--with-readline=#{Formula["readline"].opt_prefix}"
+      system "make"
+    end
+
+    # Create a symlink `bin/gap` from the `gap` binary
+    bin.install_symlink libexec/"gap" => "gap"
+
+    ohai "Building included packages. Please be patient, it may take a while"
+    cd libexec/"pkg" do
+      # NOTE: This script will build most of the packages that require
+      # compilation. It is known to produce a number of warnings and
+      # error messages, possibly failing to build several packages.
+      system "../bin/BuildPackages.sh", "--with-gaproot=#{libexec}"
```

Comment:
Note that these packages have many additional  dependencies; the package typically will refuse to build if the dependency is missing; but this means that different people using this very same recipe will get different results, depending on what stuff they have already installed on their system. I don't know much about homebrew, but in Fink and MacPorts, that would be unacceptable. Some dependency of certain GAP packages include: boost, mpfr, mpfi, normaliz, singular, polymake,libcurl, ...

Reply:
If this can be trivially resolved by several extra lines with `depends_on` then I am all for it. I can modify the Formula to display `pkg/log/fail.log` to report on packages that failed to build, and have a several iterations with Homebrews's Jenkins CI, to see what happens in a clean build.

Reply:
Well, potential dependencies include: 4ti2, boost, curl, flint, fplll, glpk, mpc, mpfi, mpfr, normaliz, ntl, pari, polymake, singular, zeromq and probably some more I forgot about.

One may wish to select a subset (e.g. because homebrew does not have something, or because it's rather big), but then I'd suggest to explicitly exclude the affected GAP package(s) from compilation.